src/models.jl

```diff
@@ -0,0 +1,124 @@
+mutable struct Model
+    metadata::ModelMetadata
+    configuration::ModelConfiguration
+    boundary_conditions::BoundaryConditions
+    constants::PlanetaryConstants
+    eos::EquationOfState
+    grid::Grid
+    velocities::VelocityFields
+    tracers::TracerFields
+    pressures::PressureFields
+    G::SourceTerms
+    Gp::SourceTerms
+    forcings::ForcingFields
+    forcing::Forcing
+    stepper_tmp::StepperTemporaryFields
+    operator_tmp::OperatorTemporaryFields
+    ssp  # ::SpectralSolverParameters or ::SpectralSolverParametersGPU
+    clock::Clock
+    output_writers::Array{OutputWriter,1}
+    diagnostics::Array{Diagnostic,1}
+end
+
+"""
+    Model(; kwargs...)
+
+Construct an `Oceananigans.jl` model. The keyword arguments are:
+
+          N (tuple) : model resolution in (x, y, z)
+          L (tuple) : model domain in (x, y, z)
+         dt (float) : time step
+ start_time (float) : start time for the simulation
+  time_step (float) : ?
+         arch (sym) : architecture (:cpu or :gpu)
+  float_type (type) : floating point type for model data (typically Float32 or Float64)
+          constants : planetary constants (?)
+                eos : equation of state to infer density from temperature and salinity
+            forcing : forcing functions for (u, v, w, T, S)
+boundary_conditions : boundary conditions
+     output_writers : output writer
+        diagonstics : diagnostics
+
+        ... and more.
+"""
```

Comment:
This is awesome, should make what the model is doing much more obvious to new users!